File: src/main/java/io/github/mg138/tsbook/config/item/element/SimpleItemSetting.java
package io.github.mg138.tsbook.config.item.element;

import io.github.mg138.tsbook.Book;
import org.bukkit.Material;
import org.bukkit.configuration.file.YamlConfiguration;

import java.util.List;

/*
    Declares the most basic instance of a ItemSetting
 */

public class SimpleItemSetting extends ItemSetting {
    public SimpleItemSetting(YamlConfiguration setting) {
        this(
                setting.getString("ID"),
                setting.getString("ITEM_TYPE"),
                Material.valueOf(setting.getString("MATERIAL")),
                setting.getInt("MODEL"),
                Book.Companion.getCfg().translate.translateString(setting.getString("NAME")),
                Book.Companion.getCfg().translate.translateByList(setting.getStringList("LORE"))
        );
    }

    public SimpleItemSetting(String id, String item_type, Material material, Integer model, String name, List<String> lore) {
        super(id, item_type, material, model, name, lore);
    }
}

File: src/main/java/io/github/mg138/tsbook/setting/item/element/SimpleItemSetting.kt
package io.github.mg138.tsbook.setting.item.element

import io.github.mg138.tsbook.setting.BookConfig
import org.bukkit.Material
import org.bukkit.configuration.file.YamlConfiguration

/*
    Declares the most basic instance of a ItemSetting
 */
class SimpleItemSetting(
    id: String,
    item_type: String,
    material: Material,
    model: Int,
    name: String,
    lore: List<String>
) : ItemSetting(id, item_type, material, model, name, lore) {
    constructor(setting: YamlConfiguration) : this(
        setting.getString("ID")!!,
        setting.getString("ITEM_TYPE")!!,
        Material.valueOf(setting.getString("MATERIAL")!!),
        setting.getInt("MODEL"),
        BookConfig.translate.translate("FORMAT.NAME", null, setting),
        BookConfig.translate.translateList("FORMAT.LORE", null, setting)
    )
}